IdentitySdkCore/IdentitySdkCore/Classes/ReachFive.swift

```diff
@@ -94,4 +94,102 @@ public class ReachFive: NSObject {
             .authWithCode(authCodeRequest: authCodeRequest)
             .flatMap({ AuthToken.fromOpenIdTokenResponseFuture($0) })
     }
+    
+    public func signupWithWebAuthn(profile: ProfileWebAuthnSignupRequest,origin: String,friendlyName: String?,viewController: UIViewController,scopes: [String]?, completion: @escaping ((Future<AuthToken, ReachFiveError>) -> Any)) {
+           let webAuthnRegistrationRequest = WebAuthnRegistrationRequest(
+               origin: origin,
+               friendlyName: friendlyName ?? "",
+               profile: profile,
+               clientId: sdkConfig.clientId
+           )
+        // get registrationOptions
+         self.reachFiveApi
+        .createWebAuthnSignupOptions(webAuthnRegistrationRequest: webAuthnRegistrationRequest)
+        .onSuccess{registrationOptions in
+       // prepare and setup the reachFiveClientFido
+       let reachFiveClientFido = ReachFiveFidoClient (viewController: viewController, origin: origin)
+       reachFiveClientFido.setupWebAuthnClient()
+       reachFiveClientFido.startRegistration(registrationOption: registrationOptions) { (webauthnSignupCredential) -> Any in
+        // start the onSignupWithWebAuthnResult func to get firstly the authenticationToken and then exchange the tkn with an access token
+        self.onSignupWithWebAuthnResult(webauthnSignupCredential: webauthnSignupCredential,scopes: scopes)
+                   { (authToken) -> Any in
+                       _ = completion(authToken)
+                   }
+                                        }
+        }.onFailure { error in
+            var messageAlert = ""
+            switch error {
+                                case .RequestError(let requestErrors):
+                                messageAlert = requestErrors.errorUserMsg!
+                                case .TechnicalError(let technicalError):
+                                    messageAlert = (technicalError.apiError?.errorUserMsg)! as String
+                               default:
+                                   messageAlert = error.localizedDescription
+                               }
+            let alert = UIAlertController(title: "Error", message:messageAlert, preferredStyle: UIAlertController.Style.alert)
+            alert.addAction(UIAlertAction(title: "Dismiss", style: UIAlertAction.Style.default, handler: nil))
+            viewController.present(alert, animated: true, completion: nil)
```

Comment:
I don't think it's the role of the *library* to present a error message to the user. Can we let our client's integrator manage the error the way they like ?